src/makeInfinityAdder.js

```diff
@@ -4,7 +4,22 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+    let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const newSum = sum;
+
+      sum = 0;
+
+      return newSum;
+    }
+    sum += num;
+
+    return adder;
+  }
+
+  return adder;
```

Comment:
Returning the `adder` function from `makeInfinityAdder` is correct and allows the user to start chaining calls immediately after creating an instance.